neuralmonkey/trainers/generic_trainer.py

```diff
@@ -21,13 +21,22 @@
 BIAS_REGEX = re.compile(r'[Bb]ias')
 
 
-# pylint: disable=too-few-public-methods,too-many-locals
+# pylint: disable=too-few-public-methods,too-many-locals,too-many-arguments
 class GenericTrainer(object):
 
     def __init__(self, objectives: List[Objective],
                  l1_weight: float = 0.0, l2_weight: float = 0.0,
                  clip_norm: Optional[float] = None, optimizer=None,
-                 global_step=None) -> None:
+                 global_step=None, var_scopes: Optional[List[str]] = None,
+                 var_collection: Optional[str] = None) -> None:
+
+        if not var_collection:
+            var_collection = tf.GraphKeys.TRAINABLE_VARIABLES
+        if not var_scopes:
+            var_scopes = [None]
+        var_lists = [tf.get_collection(var_collection, scope)
+                     for scope in var_scopes]
+        self.var_list = [var for var_list in var_lists for var in var_list]
```

Comment:
sem by se hodil komentář, že je to zploštění seznamu seznamů, mě to vždycky hrozně vyvede z míry tyhle konstrukty